### PR TITLE
Switch to using autoplugins

### PR DESCRIPTION
**Pull request by jroper**
* Converted Sonatype plugin to an Autoplugin
* Removed deprecated keys
* Renamed non sonatype prefixed keys to have sonatype prefix
* Switched to using isSnapshot to detect snapshots
* Updated README
* Made scripted test actually do something
* Updated travis build file to including running scripted

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 language: scala
-scala:
-  - 2.10.3
```
Comment (jroper) [inline .travis.yml]: Let sbt choose the scala version it wants.

File: project/Build.scala [hunk 2]
```diff
@@ -33,7 +31,7 @@ object SonatypeBuild extends Build {
 
   import xerial.sbt.Sonatype.SonatypeKeys._
 
-  lazy val buildSettings = Defaults.defaultSettings ++ releaseSettings ++ scriptedSettings ++ xerial.sbt.Sonatype.sonatypeSettings ++ Seq[Setting[_]](
```
Comment (jroper) [inline project/Build.scala]: Defaults.defaultSettings is deprecated, automatically included by JvmPlugin autoplugin.

File: README.md [hunk 3]
```diff
@@ -47,9 +47,9 @@ credentials += Credentials("Sonatype Nexus Repository Manager",
 
 ### build.sbt
 
-Import `SonatypeKeys._` and add `xerial.sbt.Sonatype.sonatypeSettings` to your sbt settings. As least you need to set the following keys:
+sbt-sonatype is an autoplugin, it will automatically configure your build.  There are a few settings though that you need to define yourself.  Import `SonatypeKeys._`, and then define the following keys:
 
-  * `profileName` 
+  * `sonatypeProfileName`
```
Comment (xerial) [inline README.md]: I like this change, too :)

File: src/main/scala/xerial/sbt/Sonatype.scala [hunk 4]
```diff
@@ -24,20 +25,18 @@ import java.io.IOException
  * Plugin for automating release processes at Sonatype Nexus
  * @author Taro L. Saito
  */
-object Sonatype extends sbt.Plugin {
+object Sonatype extends AutoPlugin {
 
   object SonatypeKeys {
     val sonatypeRepository = settingKey[String]("Sonatype repository URL")
-    val profileName = settingKey[String]("Profile name at Sonatype: e.g. org.xerial")
+    val sonatypeProfileName = settingKey[String]("Profile name at Sonatype: e.g. org.xerial")
```
Comment (jroper) [inline src/main/scala/xerial/sbt/Sonatype.scala]: Namespace to sonatype.

File: src/main/scala/xerial/sbt/Sonatype.scala [hunk 5]
```diff
@@ -45,6 +44,15 @@ object Sonatype extends sbt.Plugin {
     val sonatypeDefaultResolver = settingKey[Resolver]("Default Sonatype Resolver")
   }
 
+  object autoImport {
+    val SonatypeKeys = Sonatype.SonatypeKeys
```
Comment (jroper) [inline src/main/scala/xerial/sbt/Sonatype.scala]: Autoplugins have their autoPlugin property auto imported (not everything in the plugin).  This ensures that SonatypeKeys is available, as before.

File: src/main/scala/xerial/sbt/Sonatype.scala [hunk 6]
```diff
@@ -45,6 +44,15 @@ object Sonatype extends sbt.Plugin {
...
+  }
+
+  override def trigger = allRequirements
+
+  override def requires = JvmPlugin
```
Comment (jroper) [inline src/main/scala/xerial/sbt/Sonatype.scala]: Automatically enabled when JvmPlugin is present.

File: src/main/scala/xerial/sbt/Sonatype.scala [hunk 7]
```diff
@@ -66,13 +74,19 @@ object Sonatype extends sbt.Plugin {
...
 
   lazy val sonatypeGlobalSettings = Seq[Def.Setting[_]](
-    sonatypeDefaultResolver := { releaseResolver(version.value) },
+    sonatypeDefaultResolver := { 
+      if (isSnapshot.value) {
+        Opts.resolver.sonatypeSnapshots
+      } else {
+        Opts.resolver.sonatypeStaging
+      }
+    },
```
Comment (jroper) [inline src/main/scala/xerial/sbt/Sonatype.scala]: Switch to using isSnapshot, and built in sbt URLs.

Comment (xerial) [inline src/main/scala/xerial/sbt/Sonatype.scala]: Thanks. I didn't know about `isSnapshot` value.

**Comment by xerial** [comment]
Looks great. Thanks for the improvement. I'll release a new version.

**Comment by xerial** [comment]
Just released 0.3.0, which will be available in a few hours.

**Comment by xerial** [comment]
@jroper 
FYI. To support multi-module project at ease, I created sbt-sonatype-0.4.0, which works by only importing `Sonatype.sonatypeSettings` at `(project root)/sonatype.sbt` file: f6fcaeb6679ff059ea704851a0764548bb789d36

With this fix, no need exists to enable/disable sbt-sonatype plugin for each project. sonatypeXXX commands become available only at the top level project, and common settings (e.g., `publishTo`, etc.) will be configured at all of the modules.

**Comment by xerial** [comment]
@jroper 
Released 0.5.0, in which almost all TaskKeys become sbt commands. In this version, we no longer need to describe `import SonatypeKeys._`. And also `sonatypeRelease` command etc. runs only once even if it is used in multi-module projects.

**Pull request merged**